storagenode/console/service.go

```diff
@@ -97,117 +102,143 @@ func NewService(log *zap.Logger, consoleDB DB, bandwidth bandwidth.DB, pieceStor
 }
 
-// GetWalletAddress return wallet address of node operator
-func (s *Service) GetWalletAddress(ctx context.Context) string {
-	defer mon.Task()(&ctx)(nil)
-	return s.walletAddress
+// Satellites represents consolidated data across all satellites.
+type Satellites struct {
+	StorageDaily   []storageusage.Stamp `json:"storageDaily"`
+	BandwidthDaily []BandwidthUsed      `json:"bandwidthDaily"`
 }
 
-// GetUptime returns current storagenode uptime
-func (s *Service) GetUptime(ctx context.Context) time.Duration {
+// GetAllSatellitesData returns bandwidth and storage daily usage consolidate
+// among all satellites from the node's trust pool.
+func (s *Service) GetAllSatellitesData(ctx context.Context) (_ *Satellites, err error) {
 	defer mon.Task()(&ctx)(nil)
-	return time.Now().Sub(s.startedAt)
-}
-
-// GetStatsFromSatellite returns storagenode stats from the satellite
-func (s *Service) GetStatsFromSatellite(ctx context.Context, satelliteID storj.NodeID) (_ *reputation.Stats, err error) {
-	defer mon.Task()(&ctx)(&err)
+	from, to := date.MonthBoundary(time.Now())
 
-	return s.reputationDB.Get(ctx, satelliteID)
-}
+	bandwidthDaily, err := s.consoleDB.Bandwidth().GetDailyTotal(ctx, from, to)
+	if err != nil {
+		return nil, SNOServiceErr.Wrap(err)
+	}
 
-// GetDailyStorageUsedForSatellite returns daily SpaceUsageStamps for a particular satellite
-func (s *Service) GetDailyStorageUsedForSatellite(ctx context.Context, satelliteID storj.NodeID, from, to time.Time) (_ []storageusage.Stamp, err error) {
-	defer mon.Task()(&ctx)(&err)
+	storageDaily, err := s.storageUsageDB.GetDailyTotal(ctx, from, to)
+	if err != nil {
+		return nil, SNOServiceErr.Wrap(err)
+	}
 
-	return s.storageusageDB.GetDaily(ctx, satelliteID, from, to)
+	return &Satellites{
+		StorageDaily:   storageDaily,
+		BandwidthDaily: bandwidthDaily,
+	}, nil
 }
 
-// GetNodeID return current node id
-func (s *Service) GetNodeID(ctx context.Context) storj.NodeID {
-	defer mon.Task()(&ctx)(nil)
-	return s.kademlia.Local().Id
-}
+// VerifySatelliteID verifies if the satellite belongs to the trust pool.
+func (s *Service) VerifySatelliteID(ctx context.Context, satelliteID storj.NodeID) (err error) {
+	defer mon.Task()(&ctx)(&err)
 
-// GetVersion return current node version
-func (s *Service) GetVersion(ctx context.Context) version.Info {
-	defer mon.Task()(&ctx)(nil)
-	return s.versionInfo
-}
+	err = s.trust.VerifySatelliteID(ctx, satelliteID)
+	if err != nil {
+		return SNOServiceErr.Wrap(err)
+	}
 
-// CheckVersion checks to make sure the version is still okay, returning an error if not
-func (s *Service) CheckVersion(ctx context.Context) (err error) {
-	defer mon.Task()(&ctx)(&err)
-	return s.version.CheckVersion(ctx)
+	return nil
 }
 
-// GetSatellites used to retrieve satellites list
-func (s *Service) GetSatellites(ctx context.Context) (_ storj.NodeIDList) {
+// CheckVersion checks if node version >= minAllowedVersion.
+func (s *Service) CheckVersion(ctx context.Context) bool {
```

Comment:
I feel like this function is not necessary. Why not asking that below function directly

Reply:
Because version dependency is encapsulated so I need to expose it(for outer usage), for inner usage I am using it directly. But you are right that right now it is not used.